crates/re_renderer/src/renderer/depth_cloud.rs

```diff
@@ -276,44 +242,6 @@ impl DepthCloudDrawData {
             depth_cloud_ubo_binding_outlines,
             depth_cloud_ubo_binding_opaque
         ) {
-            let depth_texture = match &depth_cloud.depth_data {
-                DepthCloudDepthData::U16(data) => {
-                    if cfg!(target_arch = "wasm32") {
-                        // Web: manual normalization because Depth16Unorm textures aren't supported on
-                        // the web (and won't ever be on the WebGL backend, see
-                        // https://github.com/gfx-rs/wgpu/issues/3537).
-                        //
-                        // TODO(cmc): use an RG8 texture and unpack it manually in the shader instead.
-                        use itertools::Itertools as _;
-                        let dataf32 = data
-                            .as_slice()
-                            .iter()
-                            .map(|d| *d as f32 / u16::MAX as f32)
-                            .collect_vec();
-                        create_and_upload_texture(
-                            ctx,
-                            depth_cloud,
-                            dataf32.as_slice(),
-                            wgpu::TextureFormat::R32Float,
-                        )
-                    } else {
-                        // Native: We use Depth16Unorm over R16Unorm because the latter is behind a feature flag and doesn't work on WebGPU.
-                        create_and_upload_texture(
-                            ctx,
-                            depth_cloud,
-                            data.as_slice(),
-                            wgpu::TextureFormat::Depth16Unorm,
-                        )
-                    }
-                }
-                DepthCloudDepthData::F32(data) => create_and_upload_texture(
-                    ctx,
-                    depth_cloud,
-                    data.as_slice(),
-                    wgpu::TextureFormat::R32Float,
-                ),
-            };
-
```

Comment:
This would be a good place to check that the given texture is the of the right type (single component float)

Reply:
In general I think we should take care to validate the input to `re_renderer`, because the wgpu errors are always delayed, always indecipherable, and sometimes non-existent (e.g. wrong number of components in the texture is never validated afaik, because the texture sampler always returns a vec4)

Reply:
agreed to a certain extent - anything coming into re_renderer should be validated and we're not doing that enough.

But on the flipside I wouldn't go _too_ far on the internals. Wgpu does a crazy amount of corner case validation (and it has to) - we shouldn't go overboard :)